libs/@guardian/source-development-kitchen/src/react-components/ticker/useIsInView.ts

```diff
@@ -0,0 +1,79 @@
+import { useCallback, useEffect, useRef, useState } from 'react';
+
+type Options = {
+	/**
+	 * Defaults to `undefined` (falsy),
+	 * which only trigger on the first intersection.
+	 *
+	 * If `true`, trigger the hook on all intersections.
+	 */
+	repeat?: true;
+	/** Set the initial HTML Element, if known. */
+	node?: HTMLElement;
+};
+
+/**
+ * Custom hook around the `IntersectionObserver`.
+ *
+ * @returns a tuple containing `[isInView, setNode]`
+ */
+const useIsInView = (
+	options: IntersectionObserverInit & Options,
+): [boolean, React.Dispatch<React.SetStateAction<HTMLElement | null>>] => {
+	const [isInView, setIsInView] = useState<boolean>(false);
+	const [node, setNode] = useState<HTMLElement | null>(options.node ?? null);
+
+	const observer = useRef<IntersectionObserver | null>(null);
+
+	const intersectionObserverCallback =
+		useCallback<IntersectionObserverCallback>(
+			([entry]) => {
+				if (!entry) {
+					return;
+				}
+
+				if (entry.isIntersecting) {
+					setIsInView(true);
+				} else if (options.repeat) {
+					setIsInView(false);
+				}
+			},
+			[options.repeat],
+		);
+
+	const intersectionCallback = intersectionObserverCallback;
```

Comment:
Is this line doing anything now? I noticed the DCR version is using a `debounce` option which is not present here:

https://github.com/guardian/dotcom-rendering/blob/5942fea86286277cbe28e9ca3d4dec2c9b3d6524/dotcom-rendering/src/lib/useIsInView.ts#L52